overlord/snapstate/snapstate.go

```diff
@@ -543,20 +540,28 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 		prev = unlink
 	}
 
-	// find out whether this is a core boot device
-	isCoreBoot, err := isCoreBootDevice(st)
+	// we need to know some of the characteristics of the device - it is
+	// expected to always have a model/device context at this point.
+	deviceCtx, err := DeviceCtx(st, nil, nil)
```

Comment:
during a remodel this will use the old model, to use the new model we would need to pass through the context taken by the callers, it's not a problem given the questions we are asking but this probably needs a comment, same for the usages below in the other functions

Reply:
I've added comments about this now